src/os/filestore/FileStore.cc

```diff
@@ -718,6 +718,14 @@ int FileStore::statfs(struct store_statfs_t *buf0)
   }
   buf0->total = buf.f_blocks * buf.f_bsize;
   buf0->available = buf.f_bavail * buf.f_bsize;
+  // Adjust for writes pending in the journal
+  if (journal) {
+    uint64_t estimate = journal->get_journal_size_estimate();
```

Comment:
@liupan1111 It isn't so much that buf0 is out of date it is that the journal may contain data which is pending to be written to the store.  It is not up to date with respect to the writes that have been received and not yet flushed to the store.  The journal uses a fixed amount of space so it is already accounted for.  The function name get_journal_size_estimate() is misleading because it isn't the actual size of the journal file or block device, but rather the amount of unflushed data pending to be written.  This data will be consuming space when it flushes so we should reflect that in determining how much more client I/O we should accept.